### PR TITLE
Add godoc to the templates

### PR DESCRIPTION
**Pull request by varshaprasad96**
Add godoc to client templates
Closes: #5

### CODE REVIEW

File: pkg/internal/templates.go [hunk 1]
```diff
@@ -112,22 +120,27 @@ import (
...
 	cluster  logicalcluster.LogicalCluster
 	delegate {{.Name}}{{.Version}}.{{.NameUpperFirst}}{{.VersionUpperFirst}}Interface
 }
 
+// New creates a Wrapped{{.NameUpperFirst}}{{.VersionUpperFirst}} with the given logical cluster and client interface.
 func New(cluster logicalcluster.LogicalCluster, delegate {{.Name}}{{.Version}}.{{.NameUpperFirst}}{{.VersionUpperFirst}}Interface) *Wrapped{{.NameUpperFirst}}{{.VersionUpperFirst}}{
 	return &Wrapped{{.NameUpperFirst}}{{.VersionUpperFirst}}{cluster: cluster, delegate: delegate}
 }
 
+// Wrapped{{.NameUpperFirst}}{{.VersionUpperFirst}} returns the underlying RESTClient.
```
Comment (stevekuznetsov) [inline pkg/internal/templates.go:135]:
```suggestion
// RESTClient returns the underlying RESTClient.
```

File: pkg/internal/templates.go [hunk 2]
```diff
@@ -140,6 +153,7 @@ type wrapped{{.Name}} struct {
 	delegate {{.PkgName}}{{.Version}}.{{.Name}}Interface
 }
 
+// checkCluster returns a scoped context.
```
Comment (stevekuznetsov) [inline pkg/internal/templates.go:156]: This seems misleading - `check*` implies some checking is happening, no?

Comment (varshaprasad96) [reply pkg/internal/templates.go]: Ah yeah! Modified it

**Review by stevekuznetsov — APPROVED** [review]
LGTM

File: pkg/internal/templates.go [hunk 3]
```diff
@@ -140,6 +153,9 @@ type wrapped{{.Name}} struct {
 	delegate {{.PkgName}}{{.Version}}.{{.Name}}Interface
 }
 
+// checkCluster retrives the logical cluster name from the given context and checks
```
Comment (ncdc) [inline pkg/internal/templates.go:156]:
```suggestion
// checkCluster retrieves the logical cluster name from the given context and checks
```

**Comment by ncdc** [comment]
Just 1 typo / LGTM

**Comment by ncdc** [comment]
Please rebase on top of main after #9 merges 😄

**Comment by ncdc** [comment]
Included in #11

**Pull request closed without merging**